src/EditorFeatures/Test/CodeGeneration/CodeGenerationTests.VisualBasic.cs

```diff
@@ -273,6 +274,62 @@ End Class
             }
 
             [Fact, Trait(Traits.Feature, Traits.Features.CodeGeneration)]
+            public async Task AddEvent()
+            {
+                var input = "Class [|C|] \n End Class";
+                var expected = "Class C \n Public Event E As Action \n End Class";
+                await TestAddEventAsync(input, expected,
+                    codeGenerationOptions: new CodeGenerationOptions(addImports: false));
+            }
+
+            [Fact, Trait(Traits.Feature, Traits.Features.CodeGeneration)]
+            public async Task AddEventWithAccessorAndImplementsClause()
+            {
+                var input = "Class [|C|] \n End Class";
+                var expected = "Class C \n Public Custom Event E As ComponentModel.PropertyChangedEventHandler Implements ComponentModel.INotifyPropertyChanged.PropertyChanged \n AddHandler ( value As ComponentModel . PropertyChangedEventHandler ) \n End AddHandler \n RemoveHandler ( value As ComponentModel . PropertyChangedEventHandler ) \n End RemoveHandler \n RaiseEvent ( sender As Object , e As ComponentModel . PropertyChangedEventArgs ) \n End RaiseEvent \n End Event \n End Class";
```

Comment:
120 feels about right.  And yes, a lot of existing code doesn't follow this.  However, i ascribe to the position of "always make things better" as you're going along working in an area :)